### PR TITLE
feat: validate the manifest is yaml and not changed (SRX-627DCD)

### PR DESCRIPTION
**Pull request by bjoern-michaelsen-freiheit**

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer.go [hunk 1]
```diff
@@ -329,13 +330,66 @@ func (c *CreateApplicationVersion) calculateVersion(bfs billy.Filesystem) (uint6
...
+		}
+	}
+	if c.SourceRepoUrl != "" {
+		existingSourceRepoUrl, err := util.ReadFile(fs, fs.Join(releaseDir, "source_repo_url"))
+		if err != nil || string(existingSourceRepoUrl) != c.SourceRepoUrl {
+			return false
+		}
+	}
+	for env, man := range c.Manifests {
+		envDir := fs.Join(releaseDir, "environments", env)
```
Comment (bjoern-michaelsen-freiheit) [inline services/cd-service/pkg/repository/transformer.go:384]: Should we compare canonical serializations instead?

Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/transformer.go:384]: I think we should compare after "normalization by serialization".

But more importantly: if there is a diff, then we should give the user the diff (or at least the line where they diff first), so they can understand the issue - ideally in the http response.

File: services/frontend-service/pkg/handler/release.go [hunk 2]
```diff
@@ -64,6 +64,17 @@ func isAuthor(value string) bool {
...
 
+func writeReleaseResponse(w http.ResponseWriter, r *http.Request, jsonBlob []byte, err error, status int) {
+	ctx := r.Context()
+	if err != nil {
+		logger.FromContext(ctx).Error(fmt.Sprintf("error in json.Marshal of /release: %s", err.Error()))
+		http.Error(w, "", http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(status)
+	w.Write(jsonBlob)
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/pkg/handler/release.go:75]:
```suggestion
	w.Write(jsonBlob + "\n")
```

File: pkg/api/api.proto [hunk 3]
```diff
@@ -132,7 +132,42 @@ message CreateReleaseRequest {
   string displayVersion = 10;
 }
 
+message CreateReleaseResponseSuccess {
+}
```
Comment (sven-urbanski-freiheit-com) [inline pkg/api/api.proto:136]: overall fine, but actual "diff" is missing still

File: services/cd-service/pkg/repository/transformer.go (hunk 1 above)
Comment (bjoern-michaelsen-freiheit) [reply services/cd-service/pkg/repository/transformer.go:406]: done

File: pkg/api/api.proto (hunk 3 above)
Comment (bjoern-michaelsen-freiheit) [reply pkg/api/api.proto:136]: success never has a diff (because no version existed previously in that case).

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**